solenspipe/solenspipe.py

```diff
@@ -343,15 +418,65 @@ def get_kmap(self,channel,seed,lmin,lmax,filtered=True):
         return self.cache[seed][:3] if filtered else self.cache[seed][3:]
 
     def get_mv_kappa(self,polcomb,talm,ealm,balm):
+        
         # Wrapper for qfunc
         return self.qfunc(polcomb,[talm,ealm,balm],[talm,ealm,balm])
 
     def qfunc(self,alpha,X,Y):
         # Wrapper for the core falafel full-sky lensing reconstruction function
         polcomb = alpha
-        return qe.qe_all(self.px,lambda x,y: self.theory.lCl(x,y),
+        return qe.qe_all(self.px,lambda x,y: self.theory.lCl(x,y),lambda x,y:self.theory_cross.lCl(x,y),
+                         self.mlmax,Y[0],Y[1],Y[2],estimators=[polcomb],
+                         xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])[polcomb][0]
+
+    def qfunc_bh(self,alpha,X,Y,ils,blens,bhps,Alpp,A_ps):
+        polcomb=alpha
+        Tcmb = 2.726e6
+        source=qe.qe_pointsources(self.px,lambda x,y: self.theory.lCl(x,y),lambda x,y:self.theory_cross.lCl(x,y),
+                         self.mlmax,Y[0],Y[1],Y[2],estimators=[polcomb],
+                         xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])
+        phi=qe.qe_all(self.px,lambda x,y: self.theory.lCl(x,y),lambda x,y:self.theory_cross.lCl(x,y),
                          self.mlmax,Y[0],Y[1],Y[2],estimators=[polcomb],
                          xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])[polcomb][0]
+        s_alms=qe.filter_alms(source,maps.interp(ils,A_ps*bhps*Tcmb**2))
+        phi_alms = qe.filter_alms(phi,maps.interp(ils,2*Alpp*blens))
+        balms=phi_alms-s_alms
+        recon_alms=hp.almxfl(balms,ils*(ils+1)*0.5)
+        return recon_alms
+
+    def get_mv_curl(self,polcomb,talm,ealm,balm):
+    
+        return self.qfunc_curl(polcomb,[talm,ealm,balm],[talm,ealm,balm])
+
+    def qfunc_curl(self,alpha,X,Y):
```

Comment:
@qujia7 is this different from `qfunc` above?

Reply:
The only difference is that qfunc calls for the [0] 's element of the qe_all which is the phi_alms. And qfunc_curl calls for the [1] element which is the curl. The reason for two functions was thinking for the RDN0 step, to input the correct alms for the bias calculation.